plutus-ledger/src/Ledger/Tx.hs

```diff
@@ -96,16 +101,17 @@ type PrivateKey = Crypto.XPrv
 -- This datatype was created in order to be used in
 -- 'Ledger.Constraints.processConstraint', specifically with the constraints
 -- 'MustSpendPubKeyOutput' and 'MustSpendScriptOutput'.
---
--- TODO Add 'Either DatumHash Datum' field for 'PublicKeyChainIndexTxOut'.
 data ChainIndexTxOut =
-    PublicKeyChainIndexTxOut { _ciTxOutAddress :: Address
-                             , _ciTxOutValue   :: Value
+    PublicKeyChainIndexTxOut { _ciTxOutAddress         :: Address
+                             , _ciTxOutValue           :: V1.Value
+                             , _ciTxOutDatumPublicKey  :: V2.OutputDatum
```

Comment:
This reads wrong, how about `_ciTxOutPublicKeyDatum` and `_ciTxOutScriptDatum`

Reply:
Yes, it sounds better.